main.js

```diff
@@ -1 +1,79 @@
-// write your code here to make the tests pass
+var Library = function () {
+  var books = [];
+
+var addBook = function (book) {
+  books.push(book);
+};
+
+var listBooks = function() {
+  for (var i = 0; i < books.length; i++) {
+    console.log(books[i].getAttribute('title'))
+  }
+}
+
+var checkOutBook = function (book) {
+  for (var i = 0; i < books.length; i++) {
+    if (books[i].getAttribute('title') === book.getAttribute('title')) {
+      book.setAttribute('checkedOut', true)
+    }
+}
+}
+var returnBook = function(book) {
+  for (var i = 0; i < books.length; i++) {
+    if (books[i].getAttribute('title') === book.getAttribute('title')) {
+      book.setAttribute('checkedOut', false)
+    }
+
+  }
+}
+
+return {
+  addBook: addBook,
+  checkOutBook: checkOutBook,
+  returnBook: returnBook,
+  listBooks: listBooks
+}
+
+};
+var Book = function (title, author) {
+  var attributes = {
+    title: title,
+    author: author,
+    checkedOut: false
+  }
+  var getAttribute = function(attribute) {
+    var keys = Object.keys(attributes);
+    for (var i = 0; i < keys.length; i++) {
+      if (keys[i] === attribute) {
+      return attributes[attribute];
+  };
+}
+}
+  var setAttribute = function(attribute, value) {
+    var keys = Object.keys(attributes);
+    for (var i = 0; i < keys.length; i++) {
+      if (keys[i] === attribute) {
+        attributes[attribute] = value;
+      }
+    }
+  }
+
+  return {
+    getAttribute: getAttribute,
+    setAttribute: setAttribute
+  };
+};
+//create library
```

Comment:
I like the comments! Being concise and descriptive for these is super important and will help you in the long run. This combined with decent naming conventions for your functions and variables will allow your code to be much more readable to both you and others. Good job, Betty.

Reply:
Thanks!